### PR TITLE
Feature/working limit

### PR DESCRIPTION
**Pull request by BrandonSharratt**
# Description

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (non-breaking change with enhancements to documentation)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/bcgov/OCWA/blob/master/CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: microservices/validateApi/validator/validator.py [hunk 1]
```diff
@@ -10,23 +11,106 @@
...
+        else:
+            item = ValidationQueue.getQueue().get(False)
+            if item.size == -1:
+                try:
+                    headObj = conn.head_object(Bucket=bucket, Key=item.result.file_id)
+                    item.size = headObj['ContentLength']
+                except:
+                    log.error("error getting object from s3")
+                
+
```
Comment (jujaga) [inline microservices/validateApi/validator/validator.py]: Extra whitespace line?

File: microservices/validateApi/validator/validator.py [hunk 2]
```diff
@@ -10,23 +11,106 @@
...
+                item.result.save()
+
+            elif (workingSize+item.size) > workingLimit:
+                # can't work on yet, too big
+                onlyOneItem = ValidationQueue.getQueue().empty()
+                ValidationQueue.getQueue().put(item)
+                if onlyOneItem:
+                    time.sleep(SLEEP_TIME)
+            else:
```
Comment (jujaga) [inline microservices/validateApi/validator/validator.py]: Add whitespace line here

File: microservices/validateApi/ValidationQueue/ValidationQueue.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        db = Db()
+        results = db.Results.objects(state=2)
+
+        for result in results:
+            try:
+                policy = get_policy(result.rule_id)[0]
+                queueItem = QueueObject(policy, result)
+                cls.q.put(queueItem)
+            except:
+                pass
```
[the remark this answers is not in the document]
Comment (BrandonSharratt) [reply microservices/validateApi/ValidationQueue/ValidationQueue.py]: Done

**Comment by jujaga** [comment]
Proceeding with merge as this addresses Issue #99 for the validation API. Integration tests are known to fail at this time.

**Pull request merged**